src/PaginatedData.elm

```diff
@@ -1,353 +1,416 @@
-module PaginatedData
-    exposing
-        ( ContainerDict
-        , PaginatedData
-        , emptyContainer
-        , emptyPaginatedData
-        , fetchAll
-        , fetchPaginated
-        , get
-        , getAll
-        , getItemsByPager
-        , getPager
-        , getTotalCount
-        , insertDirectlyFromClient
-        , insertMultiple
-        , remove
-        , setPageAsLoading
-        , setTotalCount
-        , update
-        , viewPager
-        )
+module PaginatedData exposing
+    ( PaginatedData, PaginatedWebData, emptyPaginatedData
+    , get, getAll, getItemsByPage, getTotalCount, getPage, getLocal
+    , fetchAllPages, fetchNextPage
+    , insertLocal, handleFetchedPage, remove, setPageAsLoading, setTotalCount, update
+    , viewPager
+    )
+
+{-| By using something like
+[https://package.elm-lang.org/packages/krisajenkins/remotedata/latest](`RemoteData`),
+you can keep track of the status of a request for data: is it in progress?
+Completed? Did it fail? But what if the data comes back to you in pages?
+Keeping track of those pages is the purpose of `PaginatedData`. Which pages do
+you already have? What were their contents? Which pages do you still need? What
+requests are in flight?
+
+
+### Types
+
+@docs PaginatedData, PaginatedWebData, emptyPaginatedData
+
+
+### Accessors
+
+@docs get, getAll, getItemsByPage, getTotalCount, getPage, getLocal
+
+
+### Fetch helpers
 
-{-| A `PaginatedData` represents a dict of values, that are paginated on the
-server.
+@docs fetchAllPages, fetchNextPage
 
-@docs ContainerDict, PaginatedData, emptyContainer, emptyPaginatedData, fetchAll, fetchPaginated, get, getAll, getItemsByPager, getPager, getTotalCount, insertDirectlyFromClient, insertMultiple, remove, setPageAsLoading, setTotalCount, update, viewPager
+
+### Updaters
+
+@docs insertLocal, handleFetchedPage, remove, setPageAsLoading, setTotalCount, update
+
+
+### View
+
+@docs viewPager
 
 -}
 
-import EveryDict exposing (EveryDict)
+import Dict exposing (Dict)
 import EveryDictList exposing (EveryDictList)
 import Html exposing (Html, a, li, text, ul)
 import Html.Attributes exposing (action, class, classList)
 import Html.Events exposing (onClick)
-import RemoteData exposing (WebData)
+import Http
+import RemoteData exposing (RemoteData(..))
 
 
-{-| A container Dict can act as a local cache.
+{-| Represents the status of paged data, where we may have some or all of the
+pages, and requests for one or more pages may be in-flight or have had errors.
+You may also have some "local" data which is not on any page.
```

Comment:
Maybe also add => `You may also have some "local" data which is not on any page. That is, values that may already be saved in the backend, but you didn't re-fetch a list of items, thus do not yet know the page of those items should belong to.`